Java/core/src/main/java/com/amazon/randomcutforest/statistics/Deviation.java

```diff
@@ -84,6 +84,8 @@ public double getDiscount() {
     }
 
     public void setDiscount(double discount) {
+        checkArgument(discount >= 0, "cannot be negative");
+        checkArgument(discount < 1, "can be at most 1");
```

Comment:
Stochastic discount of 1 means that every past value other than the current value has weight 0 (or does not exist/matter) because the weights decrease vy a factor of (1-discount). A discount of 1 does not seem to apply for these time series problems.